.travis.yml

```diff
@@ -1,27 +1,48 @@
 language: c
+
 compiler:
   - gcc
+
 before_script:
+  - sudo apt-get -qq update
+  - sudo apt-get install -y valgrind
+
   - git submodule update --init --recursive
   - mkdir build
   - cd build
+
   - mkdir debug-with-coverage
   - cd debug-with-coverage
   - cmake ../.. -DCMAKE_BUILD_TYPE=DebugWithCoverage -DCMAKE_C_OUTPUT_EXTENSION_REPLACE=ON
   - cd ..
+
   - mkdir release
   - cd release
   - cmake ../.. -DCMAKE_BUILD_TYPE=Release
   - cd ..
+
+  - mkdir memcheck
+  - cd memcheck
+  - cmake ../.. -DCMAKE_BUILD_TYPE=RelWithDebInfo
+  - cd ..
+
 script:
   - cd release
   - cmake --build .
   - ctest -V
   - cd ..
+
   - cd debug-with-coverage
   - cmake --build .
   - ctest -V
   - cd ..
+
+  - cd memcheck
+  - cmake --build .
+  - ctest -V -T memcheck
+  - cd ..
+
 after_success:
   - cd debug-with-coverage
   - ../../.travis/upload-gcov-results.sh `pwd`
+  - ../../.travis/show-memcheck-results.sh
```

Comment:
Missing the parameter to the script specifying the build dir.  I guess just `pwd` like gcov results.

Reply:
Thanks, good catch. That got lost in a rebasing, and for some reason travis didn't complain when this script exited with 1.

Reply:
Ah, I didn't realize that only in the `script` stage does a non-zero return code fail the build. The `after_success` doesn't care what the return codes are (makes sense, this is "after success").